src/app-layer-detect-proto.c

```diff
@@ -686,7 +686,7 @@ static uint32_t AppLayerProtoDetectProbingParserGetMask(AppProto alproto)
         FatalError("Unknown protocol detected - %u", alproto);
     }
 
-    SCReturnUInt(1UL << (uint32_t)alproto);
+    SCReturnUInt(1 << (uint32_t)alproto);
```

Comment:
Should `AppLayerProtoDetectProbingParserGetMask` return a u64 ?
But then flow structure needs to grow 2 u32 into 2 u64...

Reply:
Because we will soon have more than 32 app-layer protocols, right ?